exchanges/binance/binance.go

```diff
@@ -268,40 +268,38 @@ func (e *Exchange) GetAggregatedTrades(ctx context.Context, arg *AggregatedTrade
 	}
 	var resp []AggregatedTrade
 	path := aggregatedTrades + "?" + params.Encode()
-	return resp, e.SendHTTPRequest(ctx,
-		exchange.RestSpotSupplementary, path, spotDefaultRate, &resp)
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpotSupplementary, path, spotDefaultRate, &resp)
 }
 
 // batchAggregateTrades fetches trades in multiple requests
-// first phase, hourly requests until the first trade (or end time) is reached
-// second phase, limit requests from previous trade until end time (or limit) is reached
-func (e *Exchange) batchAggregateTrades(ctx context.Context, arg *AggregatedTradeRequestParams, params url.Values) ([]AggregatedTrade, error) {
+// If no args.FromID is passed, requests are made intervals doubling from 10s until we get a viable starting position.
+// Once the starting set is found, we continue scanning until we have all the trades in the time window
+func (e *Exchange) batchAggregateTrades(ctx context.Context, args *AggregatedTradeRequestParams, params url.Values) ([]AggregatedTrade, error) {
```

Comment:
Seems a bit backwards. I think this functionality should be in the wrapper file and only called in `GetHistoricTrades` , and GetAggregatedTrades get called without the batch handling and the added complexity and return exactly what it needs. Just to map to API. But not worth changing.